### PR TITLE
Added ARC I2C interfae library

### PR DESCRIPTION
**Pull request by xieqi**
Support I2C controller ACI_I2C0 and ARC_I2C1 of the ARC Core.
@SidLeung

### CODE REVIEW

File: ARC_I2C/library.properties [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name=ARC_I2C
+version=1.0
+author=Intel
+maintainer=Intel
+email=dave@emutex.com
+sentence=This is a I2C libraries from the Arc core that are specific to the Randon+ hardware platform.
```
Comment (eriknyquist) [inline ARC_I2C/library.properties:6]: TinyTile, not Radon+

**Comment by eriknyquist** [comment]
Are these all new files? or are they ported from somewhere?

File: ARC_I2C/library.properties [hunk 2]
```diff
@@ -0,0 +1,11 @@
+name=ARC_I2C
+version=1.0
+author=Intel
+maintainer=Intel
+email=dave@emutex.com
```
Comment (eriknyquist) [inline ARC_I2C/library.properties:5]: Is this email address accurate?

**Comment by eriknyquist** [comment]
The file ARC_I2C.cpp is almost an exact duplicate of the existing file libraries/Wire/src/Wire.cpp; Besides the name change from "Wire" to "ARC_I2C", I can see perhaps 4-5 lines that are different.

Maybe the right approach here is to modify the existing driver so that a controller can be selected?

**Comment by eriknyquist** [comment]
The two example sketches, also, are slightly modified versions of two existing Wire library examples.
Can we re-think this? this looks like about 95% duplication to me. Let's see if we can re-use the existing Wire API.

-1

File: ARC_I2C/library.properties (hunk 1 above)
Comment (calvinatintel) [reply ARC_I2C/library.properties:6]: tinyTILE

**Review by eriknyquist — CHANGES_REQUESTED** [review]
@xieqi any updates on this?

**Comment by SidLeung** [comment]
This PR is replaced by,

https://github.com/01org/tinyTILE-Extended-Support-Library/pull/5

**Pull request closed without merging**